rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -88,39 +90,40 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
         color_format = b.getOutputColorFormat(self.loader._handle)
         self.p = (1 if (color_format == int(types.GRAY)) else 3)
         self.labels_size = ((self.bs*self.loader._numOfClasses) if (self.loader._oneHotEncoding == True) else self.bs)
-        if tensor_layout == types.NCHW:
-            if self.device == "cpu":
+        if self.tensor_format == types.NCHW:
+            if self.device == "cpu" and not self.normalization_on_device:
```

Comment:
this will break the flow when there is multiple outputs from the pipeline.

Reply:
Changed the tensor creation according to the output memory type as discussed in the call